packages/ui-react-utils/src/passthroughProps.ts

```diff
@@ -84,8 +84,10 @@ function pickProps(
   return pick(props, combined)
 }
 
-function passthroughProps(props: Record<string, any>) {
-  const validProps: Record<string, unknown> = {}
+function passthroughProps<Props extends Record<string, any>>(
+  props: Props
+): Partial<Props> {
```

Comment:
I'm not sure about this `Partial<Props>` return type here, becuase it could lead to incorrect typing on the consuming side.
For example if you have a required property in `Props` and you pass the whole props object to `passthroughProps` then that required property now will display as an optional property.
Another problem is that the `style, className, children, styles, makeStyles` will be shown as actual properties on the returned object even tough `passthroughProps` clearly filters those properties out.

I know it is really hard to type these kind of really dynamic solutions without rewriting them alltogheter, but I think we could do a type casting for the returned value, such as:

```ts
...
return validProps as Omit<Props, 'style' | 'className' | 'styles' | 'makeStyles'>
```

This will keep the required props being required and at least will not show those style and className related props.

Reply:
I reverted this change, because it was not needed for the typing of Portal, and it seems more complicated. Let's wait until we have the correct typing of Components and decorators/HOCs, and come back to this.